integration-cli/docker_cli_port_test.go

```diff
@@ -135,65 +129,89 @@ func (s *DockerSuite) TestPortList(c *testing.T) {
 		"80/tcp -> 0.0.0.0:9800",
 		"81/tcp -> 0.0.0.0:9801",
 		"82/tcp -> 0.0.0.0:9802",
-		"83/tcp -> 0.0.0.0:9803"})
+		"83/tcp -> 0.0.0.0:9803",
+	})
 	// Port list is not correct
 	assert.NilError(c, err)
 	dockerCmd(c, "rm", "-f", ID)
 
 	// test mixing protocols in same port range
-	out, _ = dockerCmd(c, "run", "-d",
-		"-p", "8000-8080:80",
-		"-p", "8000-8080:80/udp",
-		"busybox", "top")
+	out, _ = dockerCmd(c, "run", "-d", "-p", "8000-8080:80", "-p", "8000-8080:80/udp", "busybox", "top")
 	ID = strings.TrimSpace(out)
 
 	out, _ = dockerCmd(c, "port", ID)
 
 	// Running this test multiple times causes the TCP port to increment.
-	err = assertPortRange(c, out, []int{8000, 8080}, []int{8000, 8080})
+	err = assertPortRange(ID, []int{8000, 8080}, []int{8000, 8080})
 	// Port list is not correct
 	assert.NilError(c, err)
 	dockerCmd(c, "rm", "-f", ID)
 }
 
 func assertPortList(c *testing.T, out string, expected []string) error {
+	c.Helper()
 	lines := strings.Split(strings.Trim(out, "\n "), "\n")
 	if len(lines) != len(expected) {
 		return fmt.Errorf("different size lists %s, %d, %d", out, len(lines), len(expected))
 	}
 	sort.Strings(lines)
 	sort.Strings(expected)
 
+	// "docker port" does not yet have a "--format" flag, and older versions
+	// of the CLI used an incorrect output format for mappings on IPv6 addresses
+	// for example, "80/tcp -> :::80" instead of "80/tcp -> [::]:80".
+	oldFormat := func(mapping string) string {
+		old := strings.Replace(mapping, "-> [", "-> ", 1)
+		old = strings.Replace(old, "]:", ":", 1)
```

Comment:
I made the normalisation too strict in this PR / https://github.com/moby/moby/pull/42050, and forgot the "expected" format was sometimes passed as a "full" format (`80/tcp -> [::]:80`), and sometimes only the mapping (`[::]:80`); pushed a fix as a separate commit in https://github.com/moby/moby/pull/41908/files#r601203287, but I can move to a separate PR if we think that makes more sense